cylc/rose/entry_points.py

```diff
@@ -0,0 +1,207 @@
+
+    Args:
+        dir_(string or pathlib.Path):
+            Search for a ``rose-suite.conf`` file in this location.
+        dest_root (string or pathlib.Path)
+
+    """
+    if not rose_config_exists(dir_, opts):
+        return False
+
+    # Load the config tree
+    config_tree = rose_config_tree_loader(dir_, opts)
+    if any(['file' in i for i in config_tree.node.value]):
+
+        # Carry out imports.
+        from metomi.rose.config_processor import ConfigProcessorsManager
+        from metomi.rose.popen import RosePopener
+        from metomi.rose.reporter import Reporter
+        from metomi.rose.fs_util import FileSystemUtil
+
+        # Update config tree with install location
+        # NOTE-TO-SELF: value=os.environ["CYLC_SUITE_RUN_DIR"]
+        config_tree.node = config_tree.node.set(
+            keys=["file-install-root"], value=str(dest_root)
+        )
+
+        # Artificially set rose to verbose.
+        # TODO - either use Cylc Log as event handler, or get Cylc Verbosity
+        # settings to pass to Rose Reporter.
+        event_handler = Reporter(3)
+        fs_util = FileSystemUtil(event_handler)
+        popen = RosePopener(event_handler)
+
+        # Process files
+        config_pm = ConfigProcessorsManager(event_handler, popen, fs_util)
+        config_pm(config_tree, "file")
+
+    return True
+
+
+def record_cylc_install_options(
+    dest_root=None,
+    opts=None,
+    dir_=None,
+):
+    """Create/modify files recording Cylc install config options.
+
+    Creates a new config based on CLI options and writes it to the workflow
+    install location as ``rose-suite-cylc-install.conf``. If
+    ``rose-suite-cylc-install.conf`` already exists over-writes changed items,
+    except for ``!opts=`` which is merged and simplified.
+
+    If ``!opts=`` have been changed these are appended to those that have
+    been written in the installed ``rose-suite.conf``.
+
+    Args:
+        _ (pathlib.Path | or str):
+            Not used in this function, but requried for consistent entry point.
+        opts:
+            Cylc option parser object - we want to extract the following
+            values:
+            - opt_conf_keys (list or str):
+                Equivelent of ``rose suite-run --option KEY``
+            - defines (list of str):
+                Equivelent of ``rose suite-run --define KEY=VAL``
+            - suite_defines (list of str):
+                Equivelent of ``rose suite-run --define-suite KEY=VAL``
+        dest_root (pathlib.Path | or str):
+            Path to dump the rose-suite-cylc-conf
+
+    Returns:
+        cli_config - Config Node which has been dumped to
+        ``rose-suite-cylc-install.conf``.
+        rose_suite_conf['opts'] - Opts section of the config node dumped to
+        installed ``rose-suite.conf``.
+    """
+    # Construct a path objects representing our target files.
+    conf_filepath = Path(dest_root) / 'rose-suite-cylc-install.conf'
```

Comment:
To make this register with Rose as an opt config it needs to go into the `opt/` dir.

```suggestion
    conf_filepath = Path(dest_root, 'opt', 'rose-suite-cylc-install.conf'
```

Reply:
I've misunderstood this then. I didn't realize that it was meant to go in the opts folder. That makes much more sense now.